### DISCUSSION TITLE
Show Mac OS Keys on Windows

### DISCUSSION THREAD

**Discussion by Syed1337** [mulaRahul/keyviz#26] [Ideas]
Your app is awesome, I want to request a feature, I teach student computing, all my students use Mac, but I am a windows user. Is it possible for you to add a feature where Windows key such **CTRL** replaced with **Command** and so on. Really need.

**Comment by mulaRahul** [comment]
Sounds good! 

Earlier I thought about adding a feature to **show both Windows and Mac keys** like -

> Windows
> <kbd>^ Ctrl</kbd> + <kbd>S</kbd>
> Mac
> <kbd>⌘ Cmd</kbd> + <kbd>S</kbd>

so that both Windows and Mac viewers are targeted. Upon your suggestion, I think an option can be added to **show just Windows or Mac keys**. Or even **rename specific key** to something like <kbd>Win</kbd> to <kbd>Start</kbd>. 

Or **show labels** for specific shortcuts like when <kbd>Shift</kbd> + <kbd>Del</kbd> is pressed show "**Permanently Delete**" like -

> Permanently Delete
> <kbd>Shift</kbd> + <kbd>Del</kbd>

(_oops, got excited there 😋_). Well, everything aside, I'll be adding the same to future plans. Let me know if you have any more suggestions.

**Reply by Syed1337** [reply]
Please update ASAP. Class are starting from September 1st😁. Thanks a lot for such sweet contribution♥️.

**Reply by mulaRahul** [reply]
Unfortunately, I can't work on any significant or minor changes (like above) right now because the time doesn't allow me to do so 😓. 

_Alternatively_, if you're familiar with flutter you can edit the [keymaps](https://github.com/mulaRahul/keyviz/blob/main/lib/data/keymaps.dart) manually and change them accordingly like **Ctrl** to **Cmd** and so on. If you cannot do so, I can send you a makeshift patch with the said changes.

**Reply by Syed1337** [reply]
Let me try this solution. Thanks for your response

**Reply by Syed1337** [reply]
Please provide a makeshift patch with the said changes.
I'm unable to do

**Reply by mulaRahul** [reply]
Here you go [keyviz--show-mac-modifiers.zip](https://github.com/mulaRahul/keyviz/files/9393309/keyviz--show-mac-modifiers.zip)! Until the feature comes out, you can use this (_somewhat working_) patch to show <kbd>cmd</kbd> instead of <kbd>ctrl</kbd> and <kbd>option</kbd> instead of <kbd>alt</kbd> like this -

![image](https://user-images.githubusercontent.com/96373135/185895463-63ceb648-be4d-47c8-b8ae-87211c830c2a.png)

**Reply by Syed1337** [reply]
Thank you so much

**Reply by Syed1337** [reply]
Patch is not working on Win 11.
![image](https://user-images.githubusercontent.com/96580482/186485621-74bcb2a1-920f-41ed-9c5f-811f421c5105.png)

**Reply by Syed1337** [reply]
Please also add **Top Alignment** in the upcoming update. It will be easier for students to see, sitting in back seats.

**Reply by mulaRahul** [reply]
The portable version doesn't work I guess. Here is the unpacked build [keyviz--mac-patch.zip](https://github.com/mulaRahul/keyviz/files/9418290/keyviz--mac-patch.zip).

**Reply by mulaRahul** [reply]
> Please also add **Top Alignment** in the upcoming update. It will be easier for students to see, sitting in back seats.

Already in future plans https://github.com/mulaRahul/keyviz/discussions/14.

**Reply by Syed1337** [reply]
Thank You so much. It's perfectly working.
